### ISSUE TITLE
DllNotFoundException with classic Xamarin.Mac and arm64 ?

### ISSUE THREAD

**Issue by ericsink** [ericsink/SQLitePCL.raw#497]
I get the (presumably) same crash in a "classic" Xamarin.Mac app (no .NET6) - same stack trace.
```
System.DllNotFoundException has been thrown
e_sqlite3 assembly:<unknown assembly> type:<unknown type> member:(null)
```
I've reported this before [here](https://github.com/praeclarum/sqlite-net/issues/1080) with a workaround that seems to work, but it didn't get any traction.

_Originally posted by @tipa in https://github.com/ericsink/SQLitePCL.raw/issues/489#issuecomment-1161349488_

**Comment by richardschoen** [comment]
Just got the same error on a yes .Net 6.0 simple console app.

**Comment by ericsink** [comment]
@richardschoen I just wrote a simple console app using SQLitePCLRaw 2.1.0 with .NET 6.0 on M1 hardware and it worked fine.

**Comment by richardschoen** [comment]
@ericsink Are you using VS 2022 or something else like VS Code ?  Would you mind posting the project somewhere possibly ? I was also using VS 2022 for Mac

**Comment by richardschoen** [comment]
@ericsink Pretty sure it's VS.  Thanks. I'll try from the command line and VS Code and see what I get.

**Comment by ecsplendid** [comment]
I am using Unity on Macbook Pro M1 and see this error, have no idea what to do

![image](https://user-images.githubusercontent.com/6911390/181229069-0afad828-381f-4c48-80fb-47dc7cccaa97.png)

```
DllNotFoundException: e_sqlite3 assembly:<unknown assembly> type:<unknown type> member:(null)
SQLitePCL.SQLite3Provider_e_sqlite3.SQLitePCL.ISQLite3Provider.sqlite3_libversion_number () (at <7158634a0cea490799582053c6fc6ea6>:0)
SQLitePCL.raw.SetProvider (SQLitePCL.ISQLite3Provider imp) (at <00c3759168224530a91b0fc2aa365b59>:0)
DataService..ctor (System.String DatabaseName) (at Assets/Scripts/DataHandler.cs:34)
Database..ctor () (at Assets/Scripts/Database.cs:7)
Logger..ctor (System.String tag_) (at Assets/Scripts/Logger.cs:10)
AudioProcessor.Awake () (at Assets/Scripts/AudioProcessor.cs:61)
```

**Comment by ericsink** [comment]
@ecsplendid If you are using Unity, then your issue is completely separate from this one.

**Comment by tipa** [comment]
@ericsink is there anything I can do to help with this problem? Were you able to reproduce the crash?

**Comment by ericsink** [comment]
@tipa Looking back through the red herrings of this issue, I'm guessing you're talking about the original problem, which was inability to find the native library under:

> a "classic" Xamarin.Mac app (no .NET6)

Can you post a minimal repro project?

**Comment by tipa** [comment]
Sure, I can reproduce the problem with this project on a M1 Mac mini:
[SqliteTest.zip](https://github.com/ericsink/SQLitePCL.raw/files/9503221/SqliteTest.zip)

After creating an Exception breakpoint for System.Exception, the debugger should stop at `new SQLiteConnection`:
![Screenshot 2022-09-07 at 08 17 44](https://user-images.githubusercontent.com/2863198/188802973-9abc701c-4d1a-4cb7-ae94-431946560510.png)

**Comment by ericsink** [comment]
I can more or less reproduce the problem with your posted sample.  VS 2022 isn't behaving quite right for me, but it will build, and the resulting build output does not contain the `libe_sqlite3.dylib` file, which is consistent with the error you are seeing.  Also, if I run the resulting executable manually from the command line, I get the same exception.

Of course, the main question here is *why* that file isn't getting included in the build.  I don't have an answer for that yet.  It's in sqlitepclraw.lib.e_sqlite3 where I would expect it to be.

Maybe xamarin.mac never implemented support for native files stored in the runtimes folder.  Or maybe it was implemented but then it got broken during the big net6.0 transition.

**Comment by tipa** [comment]
The project above is a legacy Xamarin.Mac project, so not sure if it has something to do with .NET6.
I made another few observations: The project currently compiles and runs in x86_64 config (even on a M1), because ARM64 support is an experimental feature in legacy Xamari.Mac. When adding a reference to `SQLitePCLRaw.e_sqlite3.osx`, the project starts to work. However, when adding `<XamMacArch>x86_64,ARM64</XamMacArch>`, the app builds and runs in ARM64 on a M1 and the crash above happens again

**Comment by ericsink** [comment]
A lot of legacy Xamarin.Mac seems like black magic to me because I haven't really used it.  For example, I don't know what the `XamMacArch` msbuild property is.

It's kind of interesting that things worked when you added `SQLitePCLRaw.e_sqlite3.osx`.  That nuget package is ancient.  It's adding a very old version of SQLite.  However, that package has one very big difference:  It contains a `.targets` file that adds a `NativeReference` for the dylib.  The current setup doesn't have that, because net6.0 (I think) doesn't need it.

Perhaps I could add a similar bit of msbuild code to the current packages.  The trick will be to do so without breaking any of the net6.0 stuff.

**Comment by ericsink** [comment]
I've got a general idea of what's wrong here, but I'm not sure what the right "fix" is.

I could just do what the old package was doing, but I think that will likely make things work on Arm64 machines and break on Intel machines.

For legacy Xamarin, it's not clear exactly how the msbuild properties involved in the target CPU are supposed to work.

Since this target is going away anyway, I might try to provide something less like a fix and more like a workaround.

**Comment by tipa** [comment]
Not sure if that information helps, but I ended up adding neither the `sqlite-net-pcl` nor any `SQLitePCL.raw` NuGet package to my project but instead just copied [this source code file](https://github.com/praeclarum/sqlite-net/blob/master/src/SQLite.cs) of `sqlite-net`  into my app and things started to magically work, for both x64 and arm64

**Comment by ericsink** [comment]
Yes, that would work if you want to just use the SQLite lib provided by macOS.  There's a configuration using the nuget packages that can accomplish the same thing.

**Comment by ericsink** [comment]
Possible dupe of #358
